pandas/tests/reshape/test_merge.py

```diff
@@ -861,6 +861,14 @@ def test_validation(self):
         result = merge(left, right, on=['a', 'b'], validate='1:1')
         assert_frame_equal(result, expected_multi)
 
+    def test_merge_two_empty_df_no_division_error(self):
+        # GH17776, PR #17846
+        import warnings
```

Comment:
the test should mimic the one from the issue, except use a context manager (to avoid setting it globally)